### DISCUSSION TITLE
Max resolution?

### DISCUSSION THREAD

**Discussion by virtualbri** [bezo97/IFSRenderer#41] [General]
![2024-08-13 13 32 54](https://github.com/user-attachments/assets/915d5c4d-ca50-4c03-8837-b71aa59f5846)
Hey there, is there a max resolution?  As I increase past an 11K square, it starts to clip in the viewer until I just get a black frame by 16K.

I want to test these for some large projection format walls, and I'm seeing just how far I can push it natively.

*If* there's a config way to allow it to do larger or some settings adjustments, I'd love to know it.

Thanks!
--Brian

**Comment by bezo97** [comment]
Hi, in theory it should work on higher resolutions too, there's no maximum limit configuration. Only thing I can think of is either memory limit (VRAM), or what's more probable is that there might be some integer overflow issue when indexing the histogram buffer. On my 4070 I can go up very high, like 30K square, can confirm the clipping issue but unable to reproduce the completely black frame (that one might be out-of-memory for you).
![image](https://github.com/user-attachments/assets/05c380e5-6aae-46e8-8b15-a4188774172a)

This weekend I will try to investigate the clipping problem

**Comment by virtualbri** [comment]
I'm on a 3060 with 12GB of VRAM, but running out of it makes sense if that's the reason.  I've got 48 Gigs of regular RAM, so I'm guessing it's not that, but it could be as well.  As I increase the size more and more of the frame renders black until by 16K it's just a black frame.  

It is *computing* as far as I can tell, but doesn't display anything, and doesn't save anything, if that helps.  No error messages are returned.

**Comment by bezo97** [comment]
The problem is indeed the index overflow. I've managed to hack in a way to render a 16K*16K image correctly by splitting the histogram into 4 smaller buffers. It works but the solution is not ideal. It'll take some more time to see if there's a better way.
@virtualbri Is there a specific resolution you're aiming to use?

**Comment by virtualbri** [comment]
I wanted to test 16K, but if it's a mess, I'll just go smaller.  I just
wanted to make sure I couldn't fix something on my end to make it
possible.  I totally get that it's a ridiculously large size, so it's a big
ask. :)

I think I got it doing 10K so I'll stick with that for now.  Thank you!
